services/app-web/src/pages/Auth/cognitoAuth.ts

```diff
@@ -129,33 +130,28 @@ export async function resendSignUp(
 export async function signIn(
     email: string,
     password: string
-): Promise<CognitoUser | AmplifyError> {
+): Promise<CognitoUser | Error> {
     try {
         const result = await AmplifyAuth.signIn(email, password)
         return result.user
     } catch (e) {
         if (isAmplifyError(e)) {
             if (e.code === 'UserNotConfirmedException') {
-                console.info(
-                    'you need to confirm your account, enter the code below'
+                recordJSException(
+                    `AmplifyError ${e.code} – you need to confirm your account, enter the code below`
                 )
-                return e
             } else if (e.code === 'NotAuthorizedException') {
-                console.info('unknown user or password?')
-                return e
+                recordJSException(`AmplifyError ${e.code} – this is probably a bad password`)
             } else if (e.code === 'UserNotFoundException') {
-                console.info('user does not exist')
-                return e
+                recordJSException(`AmplifyError ${e.code} – user does not exist`)
             } else {
-                // if amplify returns an error in a format we don't expect, let's throw it for now.
-                // might be against the spirit of never throw, but this is our boundary with a system we don't control.
-                throw e
+                recordJSException(`UNEXPECTED SIGNIN ERROR AmplifyError ${e.code} – ${e.message}`)
             }
         } else {
-            console.info('didnt even get an amplify error back from login')
-            throw e
+            recordJSException(`UNEXPECTED SIGNIN ERROR – 'didnt even get an amplify error back from login`)
         }
     }
+    throw Error('Cognito SignIn error')
```

Comment:
It looks like we used to return errors when it was an error we understood and only threw when it was something truly unexpected. I don't love turning this into something that always throws. Honestly I think since writing this we've moved towards just returning un-understood errors as well

Reply:
@macrael Actually looking more at this code - all of cognitoAuth throws for errors.  The oddity was returning both Errors and throwing and thats why I moved it to throw to unify it and unify that interface on the frontend. What do you want to do here? Shall I make a ticket to refactor cognitoAuth? 

Shall I just change this one still?

Reply:
If it's not going to change having to do try/catch in our pages then I don't care if you fix this one or not, but I would like to have a ticket tracking moving this to returning errors instead. My goal is that inside any of our pages/components we never have to call try/catch because anything that can throw has been wrapped up as a dependency

Reply:
Ended up rewriting `cognitoAuth` anyway - but there are other files that have the same issue that I left alone. Looks like we already have a[ ticket](https://jiraent.cms.gov/browse/MCR-1817) to address this issue - I updated it to also include adding more eslint rules.